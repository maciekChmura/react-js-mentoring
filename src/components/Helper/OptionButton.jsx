import React, { Component } from 'react';
import styled from 'styled-components';

const Button = styled.button`
  color: white;
  height: 32px;
  margin: 4px 10px 4px 10px;
  padding: 2px 10px 2px 10px;
  font-size: 14px;
  background-color: ${props =>
    props.option === props.text ? 'salmon' : '#444'};
  border: 0;
  border-radius: 4px;
  text-transform: uppercase;
`;

class OptionButton extends Component {
  render() {
    const { changeOption, option, text } = this.props;
    return (
      <Button onClick={() => changeOption(text)} option={option} text={text}>
        {text}
      </Button>
    );
  }
}

export default OptionButton;
