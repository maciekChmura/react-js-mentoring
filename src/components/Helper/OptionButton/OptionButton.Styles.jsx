import styled from 'styled-components';

const Button = styled.button`
  color: white;
  height: 32px;
  margin: 4px 10px 4px 10px;
  padding: 2px 10px 2px 10px;
  font-size: 14px;
  background-color: ${props =>
    props.option === props.text ? 'salmon' : '#444'};
  border: 0;
  border-radius: 4px;
  text-transform: uppercase;
`;

Button.displayName = 'Button';

export default Button;
