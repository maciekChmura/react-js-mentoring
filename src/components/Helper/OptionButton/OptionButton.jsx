import React from 'react';
import Button from './OptionButton.Styles';

const OptionButton = ({ changeOption, option, text }) => (
  <Button onClick={() => changeOption(text)} option={option} text={text}>
    {text}
  </Button>
);

export default OptionButton;
