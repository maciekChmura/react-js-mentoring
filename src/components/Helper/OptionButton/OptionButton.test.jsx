import React from 'react';
import { shallow, mount } from 'enzyme';
import OptionButton from './OptionButton';

describe('OptionButton', () => {
  it('should render correctly', () => {
    const component = shallow(<OptionButton />);

    expect(component).toMatchSnapshot();
  });

  it('shoudl handle click', () => {
    const mockCallBack = jest.fn();
    const e = { preventDefault: () => {} };
    const component = mount(<OptionButton changeOption={mockCallBack} />);
    component.find('Button').prop('onClick')(e);
    expect(mockCallBack).toHaveBeenCalled();
  });
});
