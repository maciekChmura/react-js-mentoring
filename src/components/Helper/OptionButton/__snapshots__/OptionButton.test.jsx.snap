// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionButton should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "onClick": [Function],
      "option": undefined,
      "text": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "OptionButtonStyles__Button-sc-1yyses4-0",
        "isStatic": false,
        "rules": Array [
          "color:white;height:32px;margin:4px 10px 4px 10px;padding:2px 10px 2px 10px;font-size:14px;background-color:",
          [Function],
          ";border:0;border-radius:4px;text-transform:uppercase;",
        ],
      },
      "displayName": "Button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "OptionButtonStyles__Button-sc-1yyses4-0",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "onClick": [Function],
        "option": undefined,
        "text": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "OptionButtonStyles__Button-sc-1yyses4-0",
          "isStatic": false,
          "rules": Array [
            "color:white;height:32px;margin:4px 10px 4px 10px;padding:2px 10px 2px 10px;font-size:14px;background-color:",
            [Function],
            ";border:0;border-radius:4px;text-transform:uppercase;",
          ],
        },
        "displayName": "Button",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "OptionButtonStyles__Button-sc-1yyses4-0",
        "target": "button",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
